src/Surfnet/StepupSelfService/SelfServiceBundle/Service/U2fService.php

```diff
@@ -205,6 +208,69 @@ public function verifyRegistration(VerifyU2fRegistrationCommand $command)
         return RegistrationVerificationResult::success($result['status'], $result['key_handle']);
     }
 
+    public function revokeRegistration(RevokeU2fRegistrationCommand $command)
+    {
+        $this->logger->notice('Revoking U2F registration from U2F verification server');
+
+        $body = [
+            'requester' => ['institution' => $command->institution, 'identity' => $command->identityId],
+            'key_handle' => ['value' => $command->keyHandle],
+        ];
+
+        $response = $this->guzzleClient->post('api/u2f/revoke-registration', ['json' => $body, 'exceptions' => false]);
+        $statusCode = $response->getStatusCode();
+
+        try {
+            $result = $response->json();
+        } catch (\RuntimeException $e) {
+            $this->logger->critical('U2F registration revocation failed; JSON decoding failed.');
+
+            return RegistrationRevocationResult::apiError();
+        }
+
+        $hasErrors = isset($result['errors']) && is_array($result['errors'])
+            && $result['errors'] === array_filter($result['errors'], 'is_string');
+
+        if ($hasErrors && $statusCode >= 400 && $statusCode < 600) {
+            $this->logger->critical(
+                sprintf(
+                    'U2F registration revocation failed; HTTP %d with errors "%s"',
+                    $statusCode,
+                    join(', ', $result['errors'])
+                )
+            );
+
+            return RegistrationRevocationResult::apiError();
```

Comment:
why yes, yes you can :+1:
--
ibuildings
WEB & MOBILE APP DEVELOPMENT

Daan van Renterghem
Software Engineer

phone (direct): +31 30 75 51 590
skype: dvrenterghem-ibuildings
e-mail: dvrenterghem@ibuildings.nl
phone: +31 88 00 24 000
www.ibuildings.nl

To all agreements with Ibuildings.nl BV, our general terms and conditions
<http://www.ibuildings.nl/algemene_voorwaarden> apply.

On Wed, Sep 30, 2015 at 11:36 AM Reinier Kip <notifications@github.com>
wrote:

> In src/Surfnet/StepupSelfService/SelfServiceBundle/Service/U2fService.php
> <https://github.com/SURFnet/Stepup-SelfService/pull/92#discussion_r40775136>
> :
>
> > +            return RegistrationRevocationResult::apiError();
> > +        }
> > +
> > +        $hasErrors = isset($result['errors']) && is_array($result['errors'])
> > +            && $result['errors'] === array_filter($result['errors'], 'is_string');
> > +
> > +        if ($hasErrors && $statusCode >= 400 && $statusCode < 600) {
> > +            $this->logger->critical(
> > +                sprintf(
> > +                    'U2F registration revocation failed; HTTP %d with errors "%s"',
> > +                    $statusCode,
> > +                    join(', ', $result['errors'])
> > +                )
> > +            );
> > +
> > +            return RegistrationRevocationResult::apiError();
>
> Can I freely interpret this as a thumb directed away from the earth's
> centre?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SURFnet/Stepup-SelfService/pull/92/files#r40775136>.
>